src/esp/gfx/WindowlessContext.cpp

```diff
@@ -60,13 +60,30 @@ struct WindowlessContext::Impl {
     if (!windowlessGLContext_.isCreated())
       Mn::Fatal{} << "WindowlessContext: Unable to create windowless context";
 
-    makeCurrent();
+    makeCurrentPlatform();
 
     if (!magnumGLContext_.tryCreate())
       Mn::Fatal{} << "WindowlessContext: Failed to create OpenGL context";
   }
 
-  void makeCurrent() { windowlessGLContext_.makeCurrent(); }
+  void makeCurrent() {
+    makeCurrentPlatform();
+    Mn::GL::Context::makeCurrent(&magnumGLContext_);
+  }
+
+  bool makeCurrentPlatform() {
+    if (!windowlessGLContext_.makeCurrent())
+      Mn::Fatal{} << "Failed to make platform current";
```

Comment:
As far as detecting/throwing fatal errors, we also have `ESP_CHECK` and `CORRADE_ASSERT`. This style here is new to me. Is it correct?